### PR TITLE
remove 0.6-dependent code, invalid 1.0

### PR DESCRIPTION
**Pull request by matbesancon**
Clp was broken on 1.0 due to undefined initialization of arrays, also fixed various things

### CODE REVIEW

File: src/ClpCInterface.jl [hunk 1]
```diff
@@ -620,13 +620,13 @@ function number_iterations(model::ClpModel)
...
 end
-get_iteration_count = number_iterations
+const get_iteration_count = number_iterations
 
 # Set number of iterations
 function set_number_iterations(model::ClpModel, iters::Integer)
     _jl__check_model(model)
     @clp_ccall setNumberIterations Cvoid (Ptr{Cvoid}, Int32) model.p iters
-    return
+    return nothing
```
Comment (mlubin) [inline src/ClpCInterface.jl]: Use `return` instead of `return nothing`: http://www.juliaopt.org/JuMP.jl/dev/style/#Return-statements-1.

Comment (matbesancon) [reply src/ClpCInterface.jl]: ok sorry I had missed this part of the style guide

**Comment by matbesancon** [comment]
Shouldn't the test be added to `MOIT`.
The simplest way to reproduce it is an unbounded problem. With JuMP:
```julia
using JuMP; import Clp
julia> const m = Model(with_optimizer(Clp.Optimizer, LogLevel = 0))
julia> @variable(m, 0 <= x[1:5])
julia> @objective(m, Max, sum(x))
julia> optimize!(m)
ERROR: MethodError: no method matching Array{Float64,N} where N(::Int64)
Closest candidates are:
  Array{Float64,N} where N(::UndefInitializer, ::Int64) where T at boot.jl:416
  Array{Float64,N} where N(::UndefInitializer, ::Int64, ::Int64) where T at boot.jl:417
  Array{Float64,N} where N(::UndefInitializer, ::Int64, ::Int64, ::Int64) where T at boot.jl:418
  ...
Stacktrace:
 [1] unbounded_ray(::Clp.ClpCInterface.ClpModel) at /home/mbesancon/.julia/packages/Clp/zGJX2/src/ClpCInterface.jl:894

```

**Comment by odow** [comment]
Shouldn't `linear8c` have caught this?

https://github.com/JuliaOpt/MathOptInterface.jl/blob/bf36784e58ddcfcb3751e07283f8842c327f3416/src/Test/contlinear.jl#L996-L1033

**Comment by matbesancon** [comment]
there seems to be only a subset of the tests added to Clp, at least that's how I understand `test/MOI_wrapper.jl`

**Comment by odow** [comment]
Those are the excluded tests.

File: test/MOIWrapper.jl [hunk 2]
```diff
@@ -24,6 +24,7 @@ end
         # linear1 test is disabled due to the following bug:
         # https://projects.coin-or.org/Clp/ticket/84
         "linear1",
+        "linear8c",
```
Comment (mlubin) [inline test/MOIWrapper.jl]: This is the list of excluded tests. We want to run this test.

Comment (matbesancon) [reply test/MOIWrapper.jl]: oh got it

**Comment by mlubin** [comment]
So if linear8c was already running, why didn't it fail? I'm confused.

**Comment by matbesancon** [comment]
So am I, that's why I thought the argument passed was the vector of tests run, not not run

**Comment by mlubin** [comment]
linear8c was running and passing. The problematic code was run only when Clp had no ray to return:
https://github.com/JuliaOpt/Clp.jl/blob/81c2c9ade1197a5b1133a47fdc80778a6a842d9e/src/ClpCInterface.jl#L890-L895

I guess this happens in corner cases like the one you showed above where the model has no constraints. Not sure that corner case is worth a new test in MOIT. I recommend adding a small test in Clp that triggers the `else` branch.

**Comment by matbesancon** [comment]
alright, I finally managed to have a test breaking under master and passing with the fix
